src/main/signal_handlers.cpp

```diff
@@ -54,15 +55,27 @@ namespace main {
  */
 bool segvSpin = false;
 
-void print_statistics() {
-  if (pOptions != NULL && pOptions->getStatistics() && pExecutor != NULL) {
-    if (pTotalTime != NULL && pTotalTime->running()) {
+namespace signal_handlers {
+
+void print_statistics()
+{
+  if (pOptions != NULL && pOptions->getStatistics() && pExecutor != NULL)
+  {
+    if (pTotalTime != NULL && pTotalTime->running())
+    {
       pTotalTime->stop();
     }
     pExecutor->safeFlushStatistics(STDERR_FILENO);
   }
 }
 
+void timeout_handler()
+{
+  safe_print(STDERR_FILENO, "CVC4 interrupted by timeout.\n");
+  print_statistics();
+  abort();
+}
```

Comment:
It feels like it would be more consistent to move this function to src/main/time_limit.cpp (similar to `win_timeout_handler()`).